elastic/apm-server
# modelindexer: add support for gzip compression

## Motivation/summary

Add support for gzip compression to the modelindexer. Update sampling/pubsub to use modelindexer, so it can take advantage of this.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Run apm-server with `-E output.elasticsearch.experimental=true -E output.elasticsearch.compression_level=1 -E instrumentation.enabled=true`
2. Point apm-server at a proxy or some kind of fake Elasticsearch which can inspect headers and payload
3. Send events to apm-server
4. Check that payload is compressed as expected
5. Point apm-server at real Elasticsearch
6. Send more events, check that they're indexed correctly

## Related issues

Part of #6360
